src/test/ui/consts/issue-70942-trait-vs-impl-mismatch.stderr

```diff
@@ -0,0 +1,24 @@
+error[E0326]: implemented const `VALUE` has an incompatible type for trait
+  --> $DIR/issue-70942-trait-vs-impl-mismatch.rs:8:18
+   |
+LL |     const VALUE: usize;
+   |                  ----- type in trait
+...
+LL |     const VALUE: i32 = 0;
+   |                  ^^^ expected `usize`, found `i32`
+
+error[E0308]: mismatched types
+  --> $DIR/issue-70942-trait-vs-impl-mismatch.rs:13:33
+   |
+LL |     let _: [i32; Zero::VALUE] = [];
+   |            ------------------   ^^ expected `Zero::VALUE`, found `0usize`
+   |            |
+   |            expected due to this
+   |
+   = note: expected array `[i32; _]`
+              found array `[_; 0]`
```

Comment:
I don't like this error, but I'm not sure how to resolve it, unless... oh yeah I know what to do.

Reply:
The fix turned out to be a bigger refactor than anticipated, so I  opened a separate PR: #71049.